core/src/main/java/nu/yona/server/subscriptions/service/UserServiceTempEncryptionContextExecutor.java

```diff
@@ -0,0 +1,41 @@
+package nu.yona.server.subscriptions.service;
+
+import javax.transaction.Transactional;
+import javax.transaction.Transactional.TxType;
+
+import nu.yona.server.messaging.entities.MessageSource;
+import nu.yona.server.subscriptions.entities.User;
+import nu.yona.server.subscriptions.entities.UserAnonymized;
+import nu.yona.server.subscriptions.service.UserService.UserEncryptedEntitySet;
+
+import org.springframework.stereotype.Service;
+
+/*
+ * Triggers the use of new subtransactions. See
+ * http://stackoverflow.com/questions/15795985/spring-transaction-propagation-required-requires-new
+ */
+@Service
+class UserServiceTempEncryptionContextExecutor
+{
+	// use a separate transaction to commit within the crypto session
+	@Transactional(value = TxType.REQUIRES_NEW)
+	public User addUserCreatedOnBuddyRequestFlush(UserDTO buddyUserResource)
+	{
+		return User.getRepository().save(
+				User.createInstanceOnBuddyRequest(buddyUserResource.getFirstName(), buddyUserResource.getLastName(),
+						buddyUserResource.getPrivateData().getNickName(), buddyUserResource.getMobileNumber()));
+	}
+
+	// use a separate transaction to read within the crypto session
+	@Transactional(value = TxType.REQUIRES_NEW)
+	public UserEncryptedEntitySet retrieveUserEncryptedDataFlush(User originalUserEntity)
```

Comment:
First method: saves in new required transaction.
Second method: retrieves in new required transaction. Is maybe not so nice here.